### PR TITLE
mwphysics refactoring (task #5338)

### PR DESCRIPTION
**Pull request by Capostrophic**
[Task page](https://gitlab.com/OpenMW/openmw/-/issues/5338)
Put result callbacks, physics system constants, stepper and movement solver into dedicated header and implementation files, and port over some of wareya's consistency fixes in actor tracer class (make ClosestNotMeConvexResultCallback less nonsensical and use Misc::Convert functions more wherever possible).

Of course the mergeability of wareya's PR in its current state is ruined but I think we weren't going to merge that as is anyway.

And of course git blame is also ruined ~~a small price to pay for salvation~~.

I tried to minimize the chosen dependencies, but may have put more than necessary somewhere (mainly to make sure files are buildable even if they are required outside of physics system implementation).

This allows to remove 600 lines from physics system implementation.

From the user's point of view nothing has changed so I don't think updating the changelog or putting anything on .46 milestone is necessary.

### CODE REVIEW

**Comment by akortunov** [comment]
```
In file included from /home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/contacttestresultcallback.cpp:1:
/home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/contacttestresultcallback.hpp:11:1: warning: class 'btCollisionObjectWrapper' was previously declared as a struct [-Wmismatched-tags]
class btCollisionObjectWrapper;
^
/usr/include/bullet/BulletCollision/CollisionDispatch/btCollisionObjectWrapper.h:17:8: note: previous use is here
struct btCollisionObjectWrapper
       ^
/home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/contacttestresultcallback.hpp:11:1: note: did you mean struct here?
class btCollisionObjectWrapper;
^~~~~
struct
1 warning generated.
[ 85%] Building CXX object apps/openmw/CMakeFiles/openmw.dir/mwphysics/stepper.cpp.o
In file included from /home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/physicssystem.cpp:49:
/home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/contacttestresultcallback.hpp:11:1: warning: class 'btCollisionObjectWrapper' was previously declared as a struct [-Wmismatched-tags]
class btCollisionObjectWrapper;
^
/usr/include/bullet/BulletCollision/CollisionDispatch/btCollisionObjectWrapper.h:17:8: note: previous use is here
struct btCollisionObjectWrapper
       ^
/home/travis/build/OpenMW/openmw/apps/openmw/mwphysics/contacttestresultcallback.hpp:11:1: note: did you mean struct here?
class btCollisionObjectWrapper;
^~~~~
struct
1 warning generated.
```

I suppose that you use a forward declaration via "class" keyword for some structures.

File: apps/openmw/mwphysics/contacttestresultcallback.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#define OPENMW_MWPHYSICS_CONTACTTESTRESULTCALLBACK_H
+
+#include <vector>
+
+#include <BulletCollision/CollisionDispatch/btCollisionWorld.h>
+
+#include "../mwworld/ptr.hpp"
+
+class btCollisionObject;
+class btCollisionObjectWrapper;
```
Comment (akortunov) [inline apps/openmw/mwphysics/contacttestresultcallback.hpp]: I suppose that the warning come from this line.

**Comment by wareya** [comment]
> Of course the mergeability of wareya's PR in its current state is ruined but I think we weren't going to merge that as is anyway.

I did my best to explain what it did in the responses to the PR so that it wouldn't be that much of a problem if/when something like this happened anyway. Worst case scenario it can be rewritten.

**Comment by akortunov** [comment]
LGTM.

**Comment by psi29a** [comment]
Taken!

**Pull request merged**